src/gui/FileBrowser.cpp

```diff
@@ -65,17 +66,51 @@ enum TreeWidgetItemTypes
 
 
 
+void FileBrowser::addContentCheckBox()
+{
+	auto filterWidget = new QWidget(contentParent());
+	filterWidget->setFixedHeight(15);
+	auto filterWidgetLayout = new QHBoxLayout(filterWidget);
+	filterWidgetLayout->setMargin(0);
+	filterWidgetLayout->setSpacing(0);
+
+	auto configCheckBox = [this, &filterWidgetLayout](QCheckBox* box)
+	{
+		box->setObjectName("FileBrowserFilterCheckBox");
+		box->setCheckState(Qt::Checked);
+		connect(box, SIGNAL(stateChanged(int)), this, SLOT(reloadTree()));
```

Comment:
A suggestion for the unchecking both checkboxes issue. Code hasn't been tested but it should give an idea of what I mean.

```suggestion
		connect(box, SIGNAL(stateChanged(int)),
			this, [this](){
				if (!(m_showUserContent->isChecked() || m_showFactoryContent->isChecked()))
				{
					this->setChecked(true);
				}
				reloadTree();
			},
			Qt::DirectConnection
			);
```

Reply:
Thanks for the review.
> I haven't tested the build but from what I've seen at the code you can still uncheck both checkboxes but if you do all the files will show up? If so, is that preferred over having the previous behavior of just filtering everything?

No it doesn't show all files, it shows an empty directory. The following check prevents that:
```
if (!paths.isEmpty())
{
	for (QStringList::iterator it = paths.begin(); it != paths.end(); ++it)
	{
		addItems(*it);
	}
}
```

Reply:
But when you uncheck the last CheckBox, won't it call `reloadTree` again? Then I think since neither conditionals will be true (both checkboxes being unchecked) there will be no filtering of the string list.

Reply:
Oh, completely ignore the last comments, I misread the conditional :man_facepalming: 
It does behave as before hiding both user and factory folders. That snippet from before could still be used if you want to force at least one to be shown. I'm alright with the current behavior though!

Reply:
Thanks for the suggestion. I think it would be a bit surprising for the user for the following reason: Let's say that i have factory checked and user unchecked. Then if I want to have only user checked, I (could) uncheck factory first, just to be surprised that it is instantly checked again by the code above. So the only way would be to check user first and then to uncheck factory.